### ISSUE TITLE
FP S5860: consider group references through destructuring

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2737]
We should consider named capturing groups that are referenced through destructuring to avoid false-positives.

### Pattern

```javascript
const regex = /(?<foo>abc)(?<bar>xyz)/;
const found = str.match(pattern);
if (found) {
  const {foo, bar} = found.groups; // FP: unused 'foo', 'bar'
}
```

`foo` and `bar` are referenced by name through object destructuring. Therefore, they should not be reported.

### Notes

We should consider matchers from `String.prototype.match()` and `RegExp.prototype.exec()`, that is, `matcher.groups` and `matcher.indices.groups`.

### Validation

Make sure that the following issue is no longer raised:

- https://peach.sonarsource.com/project/issues?id=ghost_nodev10&issues=AXsO7S1JhneuHg3jUNkE&open=AXsO7S1JhneuHg3jUNkE
